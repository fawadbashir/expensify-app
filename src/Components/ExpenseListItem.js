import React from 'react'
import moment from 'moment'
import numeral from 'numeral'
import { Link } from 'react-router-dom'
import Card from 'react-bootstrap/Card'



const ExpenseListItem = ({ id, description, amount, createdAt }) => (


    <Card.Body style={{ borderBottom: '1px solid #e5e5e5' }} >

        <Link className='list-group-item-action' to={`/edit/${id}`}>
            <Card.Title bg='dark' as='h3'>{description}</Card.Title>
        </Link>



        <Card.Subtitle className='mt-4'> {moment(createdAt).format('MMMM do YYYY')}
            <Card.Text as='span' className='float-right '>{numeral(amount / 100).format('$0,0.00')}</Card.Text>
        </Card.Subtitle>



    </Card.Body>





)
export default ExpenseListItem