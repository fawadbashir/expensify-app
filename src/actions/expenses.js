import database from '../firebase/firebase'

//ADD_EXPENSE
export const addExpense = (expense) => ({
  type: "ADD_EXPENSE",
  expense
})

export const startAddExpense = (expenseData = {}) => {
  return async (dispatch) => {
    const {
      description = "",
      note = "",
      amount = 0,
      createdAt = 0
    } = expenseData
    const expense = { description, note, amount, createdAt }

    const ref = await database.ref('expenses').push(expense)
    dispatch(addExpense({
      id: ref.key,
      ...expense
    }))
  }
}

//REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
})

 const startRemoveExpense = ({id} = {}) => {
  return async (dispatch) => { 
 await database.ref(`expenses/${id}`).remove()
     dispatch(removeExpense({id}))
  }

}

//EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
})

//SET_EXPENSE
export const setExpenses = (expenses) => ({
  type: 'SET_EXPENSES',
  expenses
})

export const startSetExpenses = () => {
  return  async (dispatch) => {
   const snapshot = await database.ref('expenses').once('value')
    const expenses = []
    snapshot.forEach((childSnapshot) => {
      expenses.push({
        id: childSnapshot.key,
        ...childSnapshot.val()
      })
    })
    dispatch(setExpenses(expenses))
  }
}

export { startRemoveExpense }