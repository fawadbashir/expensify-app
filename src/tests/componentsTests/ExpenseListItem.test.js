import React from 'react'
import { shallow } from 'enzyme'
import {createSerializer} from 'enzyme-to-json';
import  ExpenseListItem  from '../../Components/ExpenseListItem'
import expenses from '../fixtures/expensesFixture'
expect.addSnapshotSerializer(createSerializer({mode: 'shallow'}));

test('should Render ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem  {...expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})