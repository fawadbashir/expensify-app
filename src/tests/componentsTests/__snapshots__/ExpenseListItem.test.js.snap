// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      gum
    </h3>
  </Link>
  <p>
    $1.00
     
    January 4th 1970
  </p>
</div>
`;
